### PR TITLE
sm3_mb:Add sm3_mb_test

### PR DESCRIPTION
**Pull request by haoyu01**
Add unit test without openssl dependancy

### CODE REVIEW

**Comment by haoyu01** [comment]
@gbtucker , please review

File: sm3_mb/sm3_mb_test.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+#include "sm3_mb.h"
+
+typedef struct {
+	uint8_t *msg;
+	uint32_t resultDigest[SM3_DIGEST_NWORDS];
+} TestData;
+
+static TestData test_data[] = {
+	{
+	 .msg = "abc",
```
Comment (gbtucker) [inline sm3_mb/sm3_mb_test.c:42]: Getting tons of warnings on this:
sm3_mb/sm3_mb_test.c:42:10: warning: pointer targets in initialization of ‘uint8_t *’ {aka ‘unsigned char *’} from ‘char *’ differ in signedness [-Wpointer-sign]

Comment (yuhaoth) [reply sm3_mb/sm3_mb_test.c:42]: fixed by change the pointer type at line 36

Comment (yuhaoth) [reply sm3_mb/sm3_mb_test.c:42]: done, please review

**Comment by yuhaoth** [comment]
@gbtucker this patch is rebased. please review

**Pull request merged**